src/components/Modal.js

```diff
@@ -27,11 +28,25 @@ const styles = {
 
 const Modal = ({ onDismiss, title, titleExtras, children, width = 450, showCancel = true, cancelText = 'Cancel', showX, showButtons = true, okButton, ...props }) => {
   const titleId = Utils.useUniqueId()
+  const modalNode = useRef()
+  const previouslyFocusedNode = useRef()
   // react-modal applies aria-hidden to the app root *and* takes care of limiting what can be tab-focused - see appLoader.js
   return h(RModal, {
+    contentRef: node => modalNode.current = node,
     parentSelector: () => document.getElementById('modal-root'),
     isOpen: true,
-    onRequestClose: onDismiss,
+    onRequestClose: () => {
+      onDismiss()
+      previouslyFocusedNode.current.focus()
```

Comment:
It's possible to close the modal programmatically without coming through this code path, so we should probably use the mount/unmount cycle to handle this instead.